scripts/auto-rebase/rebase_gateway_api.sh

```diff
@@ -0,0 +1,470 @@
+
+set -o errexit
+set -o errtrace
+set -o nounset
+set -o pipefail
+
+shopt -s expand_aliases
+shopt -s extglob
+
+export PS4='+ $(date "+%T.%N") ${BASH_SOURCE#$HOME/}:$LINENO \011'
+
+REPOROOT="$(readlink -f "$(dirname "${BASH_SOURCE[0]}")/../..")"
+STAGING_DIR="${REPOROOT}/_output/staging"
+PULL_SECRET_FILE="${HOME}/.pull-secret.json"
+declare -a ARCHS=("amd64" "arm64")
+declare -A GOARCH_TO_UNAME_MAP=( ["amd64"]="x86_64" ["arm64"]="aarch64" )
+
+title() {
+    echo -e "\E[34m$1\E[00m";
+}
+
+check_preconditions() {
+    if ! hash yq; then
+        title "Installing yq"
+
+        local YQ_VER=4.26.1
+        # shellcheck disable=SC2034  # appears unused
+        local YQ_HASH_amd64=9e35b817e7cdc358c1fcd8498f3872db169c3303b61645cc1faf972990f37582
+        # shellcheck disable=SC2034  # appears unused
+        local YQ_HASH_arm64=8966f9698a9bc321eae6745ffc5129b5e1b509017d3f710ee0eccec4f5568766
+        local YQ_HASH
+        YQ_HASH="YQ_HASH_$(go env GOARCH)"
+        local YQ_URL
+        YQ_URL="https://github.com/mikefarah/yq/releases/download/v${YQ_VER}/yq_linux_$(go env GOARCH)"
+        local YQ_EXE
+        YQ_EXE=$(mktemp /tmp/yq-exe.XXXXX)
+        local YQ_SUM
+        YQ_SUM=$(mktemp /tmp/yq-sum.XXXXX)
+        echo -n "${!YQ_HASH} -" > "${YQ_SUM}"
+        if ! (curl -Ls "${YQ_URL}" | tee "${YQ_EXE}" | sha256sum -c "${YQ_SUM}" &>/dev/null); then
+            echo "ERROR: Expected file at ${YQ_URL} to have checksum ${!YQ_HASH} but instead got $(sha256sum <"${YQ_EXE}" | cut -d' ' -f1)"
+            exit 1
+        fi
+        chmod +x "${YQ_EXE}" && sudo cp "${YQ_EXE}" /usr/bin/yq
+        rm -f "${YQ_EXE}" "${YQ_SUM}"
+    fi
+
+    if ! hash python3; then
+        echo "ERROR: python3 is not present on the system - please install"
+        exit 1
+    fi
+
+    if ! python3 -c "import yaml"; then
+        echo "ERROR: missing python's yaml library - please install"
+        exit 1
+    fi
+
+    if ! hash oc; then
+        echo "ERROR: oc is not present on the system - please install"
+        exit 1
+    fi
+}
+
+download_gateway_api_manifests() {
+    dest="$1"
+    version="$2"
+
+    title "downloading Gateway API manifests ${version}"
+    local GATEWAY_API_STAGING="${STAGING_DIR}/gateway-api"
+    rm -rf "${GATEWAY_API_STAGING}"
+    mkdir -p "${GATEWAY_API_STAGING}"
+
+    crd_file="${dest}/gateway.networking.k8s.io_crds.yaml"
+    oc kustomize "github.com/kubernetes-sigs/gateway-api/config/crd?ref=v${version}" > "${crd_file}"
+}
+
+# Runs each OSSM rebase step in sequence, commiting the step's output to git
+rebase_ossm_to() {
+    local operator_bundle_manifest="$1"
+    local gateway_api_version="$2"
+
+    title "# Rebasing OSSM to ${operator_bundle_manifest}"
+
+    download_ossm_operator_bundle_manifest "${operator_bundle_manifest}" "${gateway_api_version}"
+
+    # OSSM image names may include `/` and `:`, which make messy branch names.
+    rebase_branch="rebase-ossm-${operator_bundle_manifest//[:\/]/-}"
```

Comment:
Should this include the "base branch"? Like main, release-4.18 and so on? Or is it unnecessary? Will this be executed like normal rebase (periodically) or like lvms (on-demand)?

Reply:
This will be executed on demand because the bundles are artifacts from brew (they get published as catalogs, this is an optional operator).